tsl/src/nodes/decompress_chunk/exec.c

```diff
@@ -291,60 +424,116 @@ decompress_chunk_begin(CustomScanState *node, EState *estate, int eflags)
 		}
 	}
 
-	initialize_column_state(state);
-
 	node->custom_ps = lappend(node->custom_ps, ExecInitNode(compressed_scan, estate, eflags));
-
-	state->per_batch_context = AllocSetContextCreate(CurrentMemoryContext,
-													 "DecompressChunk per_batch",
-													 ALLOCSET_DEFAULT_SIZES);
 }
 
-static void
-initialize_batch(DecompressChunkState *state, TupleTableSlot *compressed_slot,
-				 TupleTableSlot *decompressed_slot)
+void
+decompress_initialize_batch(DecompressChunkState *chunk_state, DecompressBatchState *batch_state,
+							TupleTableSlot *subslot)
 {
 	Datum value;
 	bool isnull;
 	int i;
-	MemoryContext old_context = MemoryContextSwitchTo(state->per_batch_context);
-	MemoryContextReset(state->per_batch_context);
 
-	state->total_batch_rows = 0;
-	state->current_batch_row = 0;
+	Assert(batch_state->initialized == false);
 
-	for (i = 0; i < state->num_columns; i++)
+	/* Batch states can be re-used skip tuple slot creation in that case */
+	if (batch_state->compressed_slot == NULL)
 	{
-		DecompressChunkColumnState *column = &state->columns[i];
+		batch_state->compressed_slot =
+			MakeSingleTupleTableSlot(subslot->tts_tupleDescriptor, subslot->tts_ops);
+	}
+	else
+	{
+		ExecClearTuple(batch_state->compressed_slot);
+	}
+
+	ExecCopySlot(batch_state->compressed_slot, subslot);
+	Assert(!TupIsNull(batch_state->compressed_slot));
+
+	/* Batch states can be re-used, so create slots one time and re-use them when the batch
+	 * state is reused.
+	 */
+	if (batch_state->decompressed_slot_scan == NULL)
+	{
```

Comment:
Can we initialize all the reused stuff in the other function? Maybe rename them for clarity, that one will be `prepare_batch_state` and this will be `start_batch_decompression`, or something like that.

Reply:
I had also thought about it. But I came to the conclusion that we should initialize them lazily. It is not guaranteed that every `batch_state` will be used (especially if we create large quantities of them in advance as proposed here https://github.com/timescale/timescaledb/pull/5530#discussion_r1176548792).

Reply:
As suggested, I added a comment about the initialization. 

Initially, I created new tuple descriptors here. But now I pass here only the raw reference to the original tuple descriptors.